smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/ServiceBareBonesClientGenerator.java

```diff
@@ -253,11 +253,13 @@ private void generateClientDefaults() {
 
         writer.writeDocs("@public")
             .openBlock("export interface ClientDefaults\n"
-                         + "  extends Partial<__SmithyResolvedConfiguration<$T>> {", "}",
+                         + "  extends Partial<__SmithyConfiguration<$T>> {", "}",
```

Comment:
ClientDefaults is on the input side, and should not have been assigned the ResolvedConfig version of "SmithyConfiguration". This is merely a correction of that.